backend/src/main/java/ai/verta/modeldb/versioning/VersioningServiceImpl.java

```diff
@@ -663,4 +667,169 @@ public void findRepositoriesBlobs(
+      } else if (request.getAttribute().getKey().isEmpty()) {
+        errorMessage = "Attribute key not found in UpdateRepositoryAttributes request";
+      }
+
+      if (errorMessage != null) {
+        LOGGER.warn(errorMessage);
+        Status status =
+            Status.newBuilder()
+                .setCode(com.google.rpc.Code.INVALID_ARGUMENT_VALUE)
+                .setMessage(errorMessage)
+                .addDetails(Any.pack(UpdateRepositoryAttribute.Response.getDefaultInstance()))
+                .build();
+        throw StatusProto.toStatusRuntimeException(status);
+      }
+
+      repositoryDAO.updateRepositoryAttribute(request.getRepositoryId(), request.getAttribute());
+      responseObserver.onNext(UpdateRepositoryAttribute.Response.newBuilder().build());
+      responseObserver.onCompleted();
+
+    } catch (Exception e) {
+      ModelDBUtils.observeError(
+          responseObserver, e, UpdateRepositoryAttribute.Response.getDefaultInstance());
+    }
+  }
+
+  @Override
+  public void getRepositoryAttributes(
+      GetRepositoryAttributes request,
+      StreamObserver<GetRepositoryAttributes.Response> responseObserver) {
+    QPSCountResource.inc();
+    try (RequestLatencyResource latencyResource =
+        new RequestLatencyResource(ModelDBAuthInterceptor.METHOD_NAME.get())) {
+      // Request Parameter Validation
+      String errorMessage = null;
+      if (!request.hasRepositoryId()
+          && request.getAttributeKeysList().isEmpty()
+          && !request.getGetAll()) {
+        errorMessage =
+            "Repository ID and Repository attribute keys not found in GetRepositoryAttributes request";
+      } else if (!request.hasRepositoryId()) {
+        errorMessage = "Repository ID not found in GetRepositoryAttributes request";
+      } else if (request.getAttributeKeysList().isEmpty() && !request.getGetAll()) {
+        errorMessage = "Repository attribute keys not found in GetRepositoryAttributes request";
+      }
+
+      if (errorMessage != null) {
+        LOGGER.warn(errorMessage);
+        Status status =
+            Status.newBuilder()
+                .setCode(com.google.rpc.Code.INVALID_ARGUMENT_VALUE)
+                .setMessage(errorMessage)
+                .addDetails(Any.pack(GetRepositoryAttributes.Response.getDefaultInstance()))
+                .build();
+        throw StatusProto.toStatusRuntimeException(status);
+      }
+
+      List<KeyValue> attributes =
+          repositoryDAO.getRepositoryAttributes(
+              request.getRepositoryId(), request.getAttributeKeysList(), request.getGetAll());
+      responseObserver.onNext(
+          GetRepositoryAttributes.Response.newBuilder().addAllAttributes(attributes).build());
+      responseObserver.onCompleted();
+
+    } catch (Exception e) {
+      ModelDBUtils.observeError(
+          responseObserver, e, GetRepositoryAttributes.Response.getDefaultInstance());
+    }
+  }
+
+  @Override
+  public void deleteRepositoryAttributes(
```

Comment:
not required

Reply:
Remove `deleteRepositoryAttributes` endpoint